### PR TITLE
Zeus: move upload endpoint params to request body

### PR DESCRIPTION
**Pull request by marwan-at-work**
Fixes https://github.com/gomods/athens/issues/218

### CODE REVIEW

File: cmd/proxy/actions/app_proxy.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ func addProxyRoutes(app *buffalo.App, storage storage.Backend) error {
 	app.GET("/{module:.+}/@v/{version}.info", cacheMissHandler(versionInfoHandler(storage), app.Worker))
 	app.GET("/{module:.+}/@v/{version}.mod", cacheMissHandler(versionModuleHandler(storage), app.Worker))
 	app.GET("/{module:.+}/@v/{version}.zip", cacheMissHandler(versionZipHandler(storage), app.Worker))
-	app.POST("/admin/upload/{module:[a-zA-Z./]+}/{version}", uploadHandler(storage))
```
Comment (michalpristas) [inline cmd/proxy/actions/app_proxy.go]: maybe removing this restriction should be enough. i like having module and version identifiers in a path, it identifies the resource. so I would prefer finding a way how to make this work and use req body as last resort.
does this make sense?

Comment (michalpristas) [reply cmd/proxy/actions/app_proxy.go]: why not letting marawan handle this as this is wonderful first PR

Comment (robjloranger) [reply cmd/proxy/actions/app_proxy.go]: Sure no problem 😊

Comment (marwan-at-work) [reply cmd/proxy/actions/app_proxy.go]: Thanks @michalpristas and @robjloranger 😄  -- For now, I made the regex just like the other ones. @robjloranger can you point me to a regex or a doc as to how Go module import paths should be matched? thanks!

Comment (michalpristas) [reply cmd/proxy/actions/app_proxy.go]: i think this should be fine. can you also add `@v/`?

Comment (marwan-at-work) [reply cmd/proxy/actions/app_proxy.go]: @michalpristas happy to. But would that potentially confuse people to think that this path is part of Vgo's Download Protocol?

Comment (michalpristas) [reply cmd/proxy/actions/app_proxy.go]: yeah probably. and as i think about it we will not want this in prod. so maybe we should add FCB to use this enpoint only in dev

Comment (marwan-at-work) [reply cmd/proxy/actions/app_proxy.go]: Would you like me to add that check as part of this PR? Is there a global variable to indicate whether runtime is dev or prod? 

Another random question: According to the wiki, Zeus can "host private modules". So I assumed this is the endpoint to which you'd do that. So wouldn't you want it in production as well? 

Thanks!

Comment (michalpristas) [reply cmd/proxy/actions/app_proxy.go]: regarding env check: https://github.com/gomods/athens/blob/58d42a1b571d60c40313c91da3f4fd468d024dd0/pkg/config/env/go.go#L11

about private modules. in v2 we will allow uploading of custom packages, right now it should be all discoverable. so private means it sits in private VCS (private gitlab instance).
so you host your code in gitlab instance deployed within companys VPN.
your private proxy  is deployed in same VPN
so it is all visible.
another way how to understand `private` is as a private repo in github visible to organization. in this case we need to provide a way how to make this package available through proxy. we don't really have this scenario thought through. 
@arschles correct me if i'm wrong

Comment (marwan-at-work) [reply cmd/proxy/actions/app_proxy.go]: ah, then maybe once the CLI is removed, this endpoint can be removed as well? For now, I put a check on top of the route. Let me know if that's what you were looking for. Thanks!

As for `private` github repos, the proxy can just use a GitHub Token to get access to private repos. So when a company deploys the proxy, they can just provide that token and the proxy knows how to use it.

**Comment by arschles** [comment]
@marwan-at-work since we're going to remove the CLI and the upload endpoint (see https://github.com/gomods/athens/issues/220), I am going to close this

**Pull request closed without merging**